docs/telescopes/scopus.md

```diff
@@ -0,0 +1,112 @@
+# SCOPUS
+
+The DAG will be a slight modification of the snapshot data source DAGs.  DAGs will be dynamically generated for each institution. Each dynamically generated DAG will be similar to the snapshot DAGs and offload the work to a Telescope class.
+
+
+## Dag naming
+
+A dynamic dag will be generated with name:
+```
+scopus_<institution>_<optional attributes>
+```
+e.g.,
+```
+scopus_curtin
+```
+
+
+## Connection id
+
+Relying on Python elsapy libraries for establishing connections. Just need to store enough credentials and parameters to fetch for each of the frameworks.
+
+
+### Naming convention of each connection id
+
+```
+scopus_<institution>[_optionally a list of _ separated other attributes]
+```
+e.g.,
+```
+scopus_curtin
+```
+
+
+### Attributes / extra fields
+```
+api_key: (if relevant)
+start_date:  Python pendulum or datetime parseable date string
+max_record_quota: (if relevant)
+```
+
+At some point it might be worthwhile to programmatically determine the optimal date ranges to minimise API calls.
+
+
+## DAG definition
+
+There will be a slight difference compared to the snapshot DAGs. It will query Airflow for a list of connection ids relevant to this particular data source, e.g., scopus_curtin, scopus_auckland, etc. A dynamically generated DAG will be used for each institution.
+
+The template for the dynamically generated DAGs will follow the pattern in the snapshot DAG, where the heavy lifting is done by Telescope classes.
+
+Listing airflow connections can be done with
+```python
+session.query(Connection)
+```
+
+
+## DAG flow
+```
+check_airflow >> check_source_server >> download_data >> upload_raw_data >> transform_data >> upload_transformed_data >> bq_load >> cleanup
+```
+
+
+## check_airflow [maybe this can be skipped]
+
+See if https://dev.elsevier.com is contactable.
+
+
+## download_data
+
+From the start date til the most recently completed month, pull data from the data source using an adapted version of Richard’s code, and append to a file on GC containing a list of json responses. Not sure if you want timing information like you did before. Maybe also round robin multiple API keys when they exist for each query.
+
```

Comment:
If we want to have a fresh copy of the most up to date data in Scopus for a particular institution, then we need to change the approach used to fetch data. The existing code only fetches data for a particular institution and month once, this means that fields such as the number of citations, will become out of date unless you fetch all of the records for each institution and date again, repeatedly.

So I think that we should change the architecture of how the telescope works to the following:
* The DAG runs every three months (could be a longer period, e.g. six months).
* Whenever the DAG runs, it fetches the records each institution, from a set date in the past (e.g. the year 2000) to the end date of the DAG run, even if it has fetched them before.
* All of the records fetched for each institution are given a release_date (the end date of the DAG run), which can be used to partition these records into a single release.